pkg/resolution/resolver/git/resolver.go

```diff
@@ -147,19 +152,32 @@ func (r *Resolver) resolveAPIGit(ctx context.Context, params map[string]string)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create SCM client: %w", err)
 	}
+
+	// If we have fake data for the scm client, replace the scm client with a fake client with the fake data
+	if r.fakeData != nil && scmClient.Driver == scm.DriverFake {
```

Comment:
would it make more sense to pass a client in as an argument to a helper function, or store the client on the resolver or something like that? (i.e. more like the dependency injection we use elsewhere in the codebase). It feels strange to mix test functionality and real functionality in the resolver code.

Reply:
I made the deliberate choice to create the client at resolution time every time. That way we don't have to specifically watch for changes to the token secret and then create a new client, and the work I'm doing for #5487 will mean we support multiple providers, which means different clients will be needed at runtime for each client, etc. 

I guess we could have a field on the resolver that's a function for creating the client, with the default implementation being what we do now, and tests can override that?

Reply:
maybe the resolution function can accept a client creation function as an argument? kind of like [getpipelinefunc](https://github.com/tektoncd/pipeline/blob/main/pkg/reconciler/pipelinerun/pipelinerun.go#L355)

Reply:
It would need to be on the resolver, not the function, since the actual path to calling `Resolve(...)` in the tests is off in `frtesting.RunResolverReconcileTest`. Lemme try something.